app/src/main/java/com/x8bit/bitwarden/ui/platform/feature/rootnav/RootNavScreen.kt

```diff
@@ -76,15 +77,18 @@ fun RootNavScreen(
         popExitTransition = { toExitTransition()(this) },
     ) {
         splashDestination()
-        authGraph(navController)
+        authGraph(
+            startDestination = (state as? RootNavState.Auth)?.startDestination ?: LANDING_ROUTE,
```

Comment:
Can you explain how this works here?

Reply:
I think I see you are using the startDestination that you've set down below.

Would it be easier to just reuse that logic here?

```kotlin
                startDestination = if (authRepository.hasUserLoggedInOrCreatedAccount) {
                    LANDING_ROUTE
                } else {
                    WELCOME_ROUTE
                }
```

Reply:
Hmmm, I see what you mean. The other option would be to move it into the top level of the state.

That would be a much bigger update, something like this:

```kotlin
data class RootNavState(
    val authStartDestination: String,
    val navState: NavState, // This NavState class would be the sealed class that RootNavState currently is
)
```

Curious if you or @brian-livefront have any thoughts on this?

Reply:
I do like that option

Reply:
So I'm not entirely certain whether or not dynamically changing destinations in a graph is a "supported" thing. The way I picture this is that setting up the `NavHost` and its destinations is really more of a one-time thing and after that, all the dynamic logic has to do with actually navigating to and within those destinations. Personally this sounds to me more like we what need is a secondary navigation once we're in the graph, which is something we already do below.

For example, when dealing with external intents like creating a new send, we first navigate to the overall graph and then do another navigation within it:

```kotlin

            RootNavState.VaultUnlockedForNewSend -> {
                navController.navigateToVaultUnlock(rootNavOptions)
                navController.navigateToAddSend(
                    sendAddType = AddSendType.AddItem,
                    navOptions = rootNavOptions,
                )
            }
```

Is there something about what you need to do here that wouldn't already fit within that pattern?

Reply:
@shannon-livefront You would not need to recreate all that logic. Those destinations are already in the graph, so you would just navigate to them like normal. The only thing you are changing with this sort of secondary navigation is the first screen that you see. What I've posted above was done to solve exactly this problem: how do you make the "start destination" dynamic.

Reply:
I think the only other option would be to move the Carousel out of the Auth Graph and make it it's own top level screen.

this was something that I had looked into previously:
https://github.com/livefront/bitwarden-android/pull/277/files#diff-2f90874fbae4270ef31e9a410a741ddacb2a78a51211ecf22319ac5f3d24b4dbR91

Reply:
Yeah that's definitely an option too. Then you have to start worrying about cross-graph transitions so it would be something you'd need to do very carefully.